contracts/libraries/VaultLifecycleTreasury.sol

```diff
@@ -766,4 +786,158 @@ library VaultLifecycleTreasury {
         }
         return friday8am;
     }
+
+    function getNextExpiry(
+        uint256 currentExpiry, 
+        uint256 day, 
+        uint256 period, // no of days in between option selling
+        bool initial
+    )
+        internal
+        pure
+        returns (uint256 nextExpiry)
+    {   
+        if (initial) {
+            uint256 weekday = getWeekday(currentExpiry) == 0
+                ? 7
+                : getWeekday(currentExpiry);
+
+            nextExpiry = (weekday >= day
+                ? currentExpiry - (weekday - day) * 1 days + 1 weeks
+                : currentExpiry + (day - weekday) * 1 days);
+        } else if (period == 30) {
+            nextExpiry = getNextMonthExpiry(currentExpiry, day); 
+        } else if (period % 7 == 0) {
+            uint256 weekday = getWeekday(currentExpiry) == 0
+                ? 7
+                : getWeekday(currentExpiry);
+
+            nextExpiry = (weekday >= day
+                    ? currentExpiry - (weekday - day) * 1 days
+                    : currentExpiry + (day - weekday) * 1 days)
+                + (period / 7) * 1 weeks;
+        }
+
+        nextExpiry = nextExpiry - (nextExpiry % (24 hours)) + (8 hours);
+        
+        return nextExpiry;
+    }
+
+    /************************************************
+     *  DATE HELPERS
+     ***********************************************/
+
+    uint constant DAY_IN_SECONDS = 86400;
+    uint constant YEAR_IN_SECONDS = 31536000;
+    uint constant LEAP_YEAR_IN_SECONDS = 31622400;
+
+    uint constant HOUR_IN_SECONDS = 3600;
+    uint constant MINUTE_IN_SECONDS = 60;
+
+    uint16 constant ORIGIN_YEAR = 1970;
+
+    function isLeapYear(uint16 year) public pure returns (bool) {
+        if (year % 4 != 0) {
+                return false;
+        }
+        if (year % 100 != 0) {
+                return true;
+        }
+        if (year % 400 != 0) {
+                return false;
+        }
+        return true;
+    }
+
+    function leapYearsBefore(uint year) public pure returns (uint) {
+        year -= 1;
+        return year / 4 - year / 100 + year / 400;
+    }
+
+    function getDaysInMonth(uint8 month, uint16 year) public pure returns (uint8) {
+        if (month == 1 || month == 3 || month == 5 || month == 7 || month == 8 || month == 10 || month == 12) {
+                return 31;
+        }
+        else if (month == 4 || month == 6 || month == 9 || month == 11) {
+                return 30;
+        }
+        else if (isLeapYear(year)) {
+                return 29;
+        }
+        else {
+                return 28;
+        }
+    }
+
+    function getNextMonthExpiry(uint256 timestamp, uint256 day) internal pure returns (uint256 nextMonthExpiry) {
+        uint secondsAccountedFor = 0;
+        uint buf;
+        uint8 i;
+        uint8 month;
+
+        // Year
```

Comment:
Usually when you have to add comments, it's a sign that it should be encapsulated into a private function.
Could we make this block something like `private getExpiryInSecondsForYear()`?

And then do this for seconds and months?

Reply:
The year, month and day are using a common variable which is secondsAccountedFor. I'm not entirely sure how this works but this variable is changed throughout to determine year, month and day